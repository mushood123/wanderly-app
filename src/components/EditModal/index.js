import { EditModal } from './editModal';

export { EditModal };
