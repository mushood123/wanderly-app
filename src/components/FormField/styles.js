import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  emptyContainer: {
    borderRadius: 10,
    padding: 20,
    borderWidth: 2,
    borderColor: 'red',
  },
  container: {
    borderRadius: 10,
    padding: 20,
    borderWidth: 1,
    borderColor: '#FFFFFF',
  },
  text: {
    fontSize: 18,
    color: 'white',
  },
  textInput: {
    fontSize: 18,
    color: '#294f8e',
  },
});
