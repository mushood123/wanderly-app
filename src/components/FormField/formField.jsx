import {TextInput} from 'react-native';
import React from 'react';
import {styles} from './styles';
import LinearGradient from 'react-native-linear-gradient';

export const FormField = ({
  title,
  style,
  secureTextEntry,
  keyboardType,
  isValidate,
  handleOnChangeText,
  value,
}) => {
  return (
    <LinearGradient
      start={{x: 0.2, y: 0}}
      end={{x: 1, y: 0}}
      colors={['#c7ceec', '#8a98e3']}
      style={{
        ...(isValidate ? styles.emptyContainer : styles.container),
        ...style,
      }}>
      <TextInput
        value={value}
        placeholderTextColor={'#294f8e90'}
        style={styles.textInput}
        placeholder={title}
        secureTextEntry={secureTextEntry}
        keyboardType={keyboardType}
        onChangeText={handleOnChangeText}
      />
    </LinearGradient>
  );
};
