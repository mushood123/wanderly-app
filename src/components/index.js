import {FormField} from './FormField';
import {Text} from './Text';

export {FormField, Text};
