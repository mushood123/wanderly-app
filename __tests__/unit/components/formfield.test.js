import React from 'react';
import { render } from '../utils/theme-wrapper/theme-wrapper';
import { FormField } from '~src/components/FormField';


describe('FormField component', () => {
    it('renders correctly and matches snapshot', () => {
        const tree = render(
            <FormField
                title="Email"
                secureTextEntry={false}
                keyboardType="default"
                isValidate={true}
                handleOnChangeText={() => { }}
                value=""
                disable={false}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with secureTextEntry', () => {
        const tree = render(
            <FormField
                title="Password"
                secureTextEntry={true}
                keyboardType="default"
                isValidate={false}
                handleOnChangeText={() => { }}
                value="password"
                disable={false}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
        const tree = render(
            <FormField
                title="Email"
                secureTextEntry={false}
                keyboardType="default"
                isValidate={true}
                handleOnChangeText={() => { }}
                value="test@test.com"
                disable={true}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
