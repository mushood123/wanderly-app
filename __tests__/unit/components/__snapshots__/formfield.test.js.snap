// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component renders correctly and matches snapshot 1`] = `
<BVLinearGradient
  colors={
    [
      4291284716,
      4287273187,
    ]
  }
  endPoint={
    {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.2,
      "y": 0,
    }
  }
  style={
    {
      "borderColor": "red",
      "borderRadius": 10,
      "borderWidth": 2,
      "padding": 20,
    }
  }
>
  <TextInput
    editable={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#294f8e90"
    secureTextEntry={false}
    style={
      {
        "color": "#294f8e",
        "fontSize": 18,
      }
    }
    value=""
  />
</BVLinearGradient>
`;

exports[`FormField component renders correctly when disabled 1`] = `
<BVLinearGradient
  colors={
    [
      4291284716,
      4287273187,
    ]
  }
  endPoint={
    {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.2,
      "y": 0,
    }
  }
  style={
    {
      "borderColor": "red",
      "borderRadius": 10,
      "borderWidth": 2,
      "padding": 20,
    }
  }
>
  <TextInput
    editable={false}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#294f8e90"
    secureTextEntry={false}
    style={
      {
        "color": "#294f8e",
        "fontSize": 18,
      }
    }
    value="test@test.com"
  />
</BVLinearGradient>
`;

exports[`FormField component renders correctly with secureTextEntry 1`] = `
<BVLinearGradient
  colors={
    [
      4291284716,
      4287273187,
    ]
  }
  endPoint={
    {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.2,
      "y": 0,
    }
  }
  style={
    {
      "borderColor": "#FFFFFF",
      "borderRadius": 10,
      "borderWidth": 1,
      "padding": 20,
    }
  }
>
  <TextInput
    editable={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="#294f8e90"
    secureTextEntry={true}
    style={
      {
        "color": "#294f8e",
        "fontSize": 18,
      }
    }
    value="password"
  />
</BVLinearGradient>
`;
